Source/WebCore/testing/MockGamepadProvider.h

```diff
@@ -59,6 +61,7 @@ class MockGamepadProvider : public GamepadProvider {
     void gamepadInputActivity();
 
     Vector<PlatformGamepad*> m_connectedGamepadVector;
+    HashMap<GamepadProviderClient*, HashSet<PlatformGamepad*>>  m_invisibleGamepadsForClient;
```

Comment:
Please use WeakHashMap and WeakHashSet instead

Reply:
@rniwa The classes in question here aren't subclasses of `EventTarget`, so I don't think these can be used in the WeakMap or Set directly, no? If we were to do that, it should probably be a separate radar?

Reply:
Also, a lot of other classes track these objects in non weak containers, for example, see: GameControllerGamepadProvider

So we should probably defer moving to weak until we can migrate everything

Reply:
Doing it in a separate PR sounds good to me.